api/clients/v2/examples/client_construction.go

```diff
@@ -0,0 +1,281 @@
+	}
+
+	relayClient, err := createRelayClient(logger, ethClient, reader.GetRelayRegistryAddress())
+	if err != nil {
+		return nil, fmt.Errorf("create relay client: %w", err)
+	}
+
+	kzgVerifier, err := createKzgVerifier()
+	if err != nil {
+		return nil, fmt.Errorf("create kzg verifier: %w", err)
+	}
+
+	relayPayloadRetrieverConfig := payloadretrieval.RelayPayloadRetrieverConfig{
+		PayloadClientConfig: *clients.GetDefaultPayloadClientConfig(),
+		RelayTimeout:        5 * time.Second,
+	}
+
+	return payloadretrieval.NewRelayPayloadRetriever(
+		logger,
+		rand.New(rand.NewSource(time.Now().UnixNano())),
+		relayPayloadRetrieverConfig,
+		relayClient,
+		kzgVerifier.Srs.G1)
+}
+
+func createValidatorPayloadRetriever() (*payloadretrieval.ValidatorPayloadRetriever, error) {
+	logger, err := createLogger()
+	if err != nil {
+		return nil, fmt.Errorf("create logger: %w", err)
+	}
+
+	// Create an EthClient for blockchain interaction
+	ethClient, err := createEthClient(logger)
+	if err != nil {
+		return nil, fmt.Errorf("create eth client: %w", err)
+	}
+
+	// Create the eth reader
+	ethReader, err := createEthReader(logger, ethClient)
+	if err != nil {
+		return nil, fmt.Errorf("create eth reader: %w", err)
+	}
+
+	chainState := eth.NewChainState(ethReader, ethClient)
+	kzgVerifier, err := createKzgVerifier()
+	if err != nil {
+		return nil, fmt.Errorf("create kzg verifier: %w", err)
+	}
+
+	// Create the retrieval client for fetching blobs from DA nodes
+	retrievalClient := clients.NewRetrievalClient(
+		logger,
+		ethReader,
+		chainState,
+		kzgVerifier,
+		10, // Number of concurrent connections to validators
+	)
+
+	// Create the ValidatorPayloadRetriever config
+	validatorPayloadRetrieverConfig := payloadretrieval.ValidatorPayloadRetrieverConfig{
+		PayloadClientConfig: *clients.GetDefaultPayloadClientConfig(),
+		RetrievalTimeout:    30 * time.Second,
+	}
+
+	return payloadretrieval.NewValidatorPayloadRetriever(
+		logger,
+		validatorPayloadRetrieverConfig,
+		retrievalClient,
+		kzgVerifier.Srs.G1)
+}
+
+func createRelayClient(
+	logger logging.Logger,
+	ethClient common.EthClient,
+	relayRegistryAddress gethcommon.Address,
+) (clients.RelayClient, error) {
+	config := &clients.RelayClientConfig{
+		UseSecureGrpcFlag:  true,
+		MaxGRPCMessageSize: 100 * 1024 * 1024, // 100 MB message size limit
+	}
+
+	relayUrlProvider, err := relay.NewRelayUrlProvider(ethClient, relayRegistryAddress)
+	if err != nil {
+		return nil, fmt.Errorf("create relay url provider: %w", err)
+	}
+
+	return clients.NewRelayClient(
+		config,
+		logger,
+		relayUrlProvider)
+}
+
+func createDisperserClient(privateKey string, kzgProver *prover.Prover) (clients.DisperserClient, error) {
+	signer, err := auth.NewLocalBlobRequestSigner(privateKey)
+	if err != nil {
+		return nil, fmt.Errorf("create blob request signer: %w", err)
+	}
+
+	disperserClientConfig := &clients.DisperserClientConfig{
+		Hostname:          "disperser-testnet-holesky.eigenda.xyz",
```

Comment:
move to top, in case they want to run against mainnet for example

Reply:
[0986099d](https://github.com/Layr-Labs/eigenda/pull/1454/commits/0986099d160c376e581ae0235185b1889f2a270b)